### PR TITLE
Send creators a weekly digest of how their games are doing

### PR DESCRIPTION
**Pull request by gtanczyk**
## Why

IL-2's last non-studio item. `POST /api/internal/digest-sweep` — one notification per creator per ISO week, riding the existing notification seam so it reaches the in-app bell, email and Web Push with no new delivery machinery, and no second place for "respect the unsubscribe" to be got wrong.

## Built from scorecards, not raw events

The nightly sweep has already paid for the aggregation, so a digest is a handful of document reads per creator rather than a telemetry scan per creator. It also means **the digest and the studio cannot disagree about a number** — both read the same document.

## Two silences, which are the design

**No evidence, no digest.** A creator whose games nobody played gets nothing, not a message full of zeros. Zeros here would be manufactured evidence of exactly the kind the rest of this system avoids — and a weekly "0 sessions" is a reason to stop opening digests at all.

**Nothing changed, no digest.** A rolling 28-day window barely moves week to week, so identical numbers are suppressed. The comparison is the **previous digest's own params**: the notification already *is* the record of what the creator was last told, and a second copy of that fact is a second thing that can drift from it.

## No feedback themes travel this path

The digest carries counts only. Themes are player-written text summarized by a model — safe to render to a signed-in creator in the studio, where they are labelled as such, and not something to push into an inbox stripped of that context and forwarded onward. The digest reports how many notes arrived and links to where they're shown.

## What the type system caught

Adding `creator.digest` to `NotificationType` failed the build in two places, both of which assumed a notification is about one submission:

- `SHORT_TYPE` builds ids as `sub-<issue>-<event>`
- `notificationCopy` renders `“«title»” <lead>`

A digest has no issue number and no single title, so rendering it through that shape would have produced `“” your games…`. Both are now keyed by a derived `SubmissionNotificationType = Exclude<NotificationType, 'creator.digest'>`, and the digest has its own email and push copy. A fourth *submission* event still picks up the existing copy automatically; a second non-submission event has to be thought about.

## Provisioning

Closed until configured, like every internal sweep — its own audience, because the audience is the endpoint's own URL:

```bash
SWEEP_URL="$(gcloud run services describe gamedev-app --region europe-west1 \
  --project gamedevpl --format 'value(status.url)')/api/internal/digest-sweep"
SA=notify-sweep@gamedevpl.iam.gserviceaccount.com
# Redeploy with DIGEST_SWEEP_AUDIENCE="$SWEEP_URL" set, then:
gcloud scheduler jobs create http digest-sweep --location europe-west1 --project gamedevpl \
  --schedule '0 9 * * 1' --uri "$SWEEP_URL" --http-method POST \
  --oidc-service-account-email "$SA" --oidc-token-audience "$SWEEP_URL"
```

Until then the endpoint exists and 401s everyone, so **merging this sends nothing**.

## Note on the link target

A digest spans every game a creator owns, so it can't deep-link to one. It points at `/` today; with #236 merged, `DIGEST_LINK` should become `/studio` — it's one constant, deliberately.

## Test plan

- [x] `npm run lint`
- [x] `npm run type-check` (api + web)
- [x] `npm test` — api 782 (+20), web 314
- [x] `npm run build`

Rebased onto master **after #236 merged**; one plan-doc conflict resolved keeping both entries.

New tests: ISO week keying including the year-boundary case that would collide two digests; no digest for an empty week; no digest when nothing moved; a digest again once something did; idempotent within a week; automation accounts skipped; one creator's failure doesn't cost the rest theirs.

## Instrumentation definition-of-done

No new capture. This is a read of the scorecards IL-2 already produces, delivered rather than waited for.

---
_Generated by [Claude Code](https://claude.ai/code/session_01JU4qCA3ruvWreLtyibx71q)_

### CODE REVIEW

File: apps/api/src/digest.ts [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+  const { store } = deps;
+  const now = deps.now ?? Date.now;
+  const currentTime = now();
+  const weekKey = isoWeekKey(currentTime);
+
+  const published = (await store.listRecentlyPublished(MAX_SUBMISSIONS_SAMPLED)).filter(
+    // Automation accounts are excluded for the same reason the creator metrics exclude
+    // them: a digest addressed to a test harness is noise with a delivery cost.
+    (submission) => !submission.ownerUid.startsWith(BOT_UID_PREFIX),
+  );
```
[the remark this answers is not in the document]
Comment (gtanczyk) [reply apps/api/src/digest.ts]: Right, and it fails in the worse of the two possible directions — losing digests rather than merely sending extra ones. Worse still, the person it drops is precisely the one the feature is for: the creator of an older game that people are *still playing*.

Inverted rather than patched. The sweep now walks **scorecards** — the evidence it reports on — and resolves each back to its owner via `getSubmissionBySlug`, instead of picking creators out of `listRecentlyPublished` and hoping their scorecards turn up. Publication date no longer has any say in who hears from us, and the "no scorecards, no digest" contract is now structural rather than something the enumeration has to remember to honour: a game with no scorecard is never enumerated, because it could not have produced a line anyway.

Bonus from the inversion: it's one fewer read per creator (no per-slug `getScorecard`), and the sampling ceiling is now reported as `truncated` instead of silently dropping the tail.

New test pins it — a game published in 2024 with recent play still produces a digest.

Your second comment is fixed in the same commit. I raised the previous-digest scan from 50 to 200 rather than removing the bound, and documented the reasoning, because the failure mode decides the size: overrunning it makes an unchanged week look like a *first* digest, so the creator receives **one duplicate, never a miss**. Between two digests a creator accrues at most a few notifications per submission, so passing 200 would take dozens of builds in a single week. A cheap, unbounded alternative would need a store query filtered by notification type, which doesn't exist yet and isn't worth adding for a failure that sends one extra email.

---
_Generated by [Claude Code](https://claude.ai/code)_
